### PR TITLE
perf: handle engine API range request in a new task

### LINKED ISSUE #3674: Handle engine API range request in a new task
### Describe the feature

this is a block operation:

https://github.com/paradigmxyz/reth/blob/42a824cf95e3cfdf2a76efe90bc65daf9012f267/crates/rpc/rpc-engine-api/src/engine_api.rs#L165-L190

which means other requests by the CL are not processed while this is handled

## TODO

* give the engine `EngineApi` a `Box< dyn TaskExecutor>`
* spawn this request onto a new task

might even redesign the `EngineApi` into `struct EngineApi {inner: Arc<EngineApiInner>}` so it's easily clonable

See for reference:

https://github.com/paradigmxyz/reth/blob/42a824cf95e3cfdf2a76efe90bc65daf9012f267/crates/rpc/rpc/src/eth/filter.rs#L28-L32

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jaylmiller**
Hello there, first time contributing. LMK what you think!

Closes #3674

### CODE REVIEW

File: crates/rpc/rpc-engine-api/src/engine_api.rs [hunk 1]
```diff
@@ -25,13 +26,15 @@ const MAX_PAYLOAD_BODIES_LIMIT: u64 = 1024;
 /// functions in the Execution layer that are crucial for the consensus process.
 pub struct EngineApi<Provider> {
     /// The provider to interact with the chain.
-    provider: Provider,
+    provider: Arc<Provider>,
```
Comment (mattsse) [inline crates/rpc/rpc-engine-api/src/engine_api.rs:29]: the assumption here is already that the `EngineApi` and hence the `Provider` are `Clone` so this isn't required.

what we should do here, what we already should have before, is to move all fields into a inner type that we then wrap in an arc

```rust
pub struct EngineApi<Provider> {
  inner: Arc<EngineApiInner<Provider>>
}

struct EngineApiInner<Provider> {
  // This is a now basically the same as EngineAPi 
}
```

all fields are very cheap to clone but they are cloned per request, so we can reduce this to one Arc::clone like this

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome, thats for taking this on, the function changes look good.

I'd like to move all fields into an Arc wrapper,

we could also do this in a follow PR, but then the Provider should still not be in an Arc here and instead restricted to be `Clone`

File: crates/rpc/rpc-engine-api/src/engine_api.rs (hunk 1 above)
Comment (jaylmiller) [reply crates/rpc/rpc-engine-api/src/engine_api.rs:29]: ok that makes sense! thanks for feedback

**Review by mattsse — APPROVED** [review]
cool!

tysm for this

**Pull request merged**